inlanecruising_plugin/src/inlanecruising_plugin.cpp

```diff
@@ -228,108 +229,99 @@ std::vector<cav_msgs::TrajectoryPlanPoint> InLaneCruisingPlugin::compose_traject
-      for (size_t i = 0; i < discreet_curve.points.size() - 1; i++)
-      {
-        Eigen::Isometry2d point_in_map =
-            curvePointInMapTF(discreet_curve.frame, discreet_curve.points[i].point, final_yaw_values.back());
-        all_sampling_points.push_back(point_in_map.translation());
-        final_yaw_values.push_back(final_yaw_values.back());
-        final_actual_speeds.push_back(final_actual_speeds.back());
-      }
-      continue;
+  std::vector<double> speed_limits;
+  std::vector<lanelet::BasicPoint2d> curve_points;
+  splitPointSpeedPairs(time_bound_points, &curve_points, &speed_limits);
+  
+  std::unique_ptr<smoothing::SplineI> fit_curve = compute_fit(curve_points); // Compute splines based on curve points
+  if (!fit_curve)
+  {  // TODO how better to handle this case
+    ROS_ERROR_STREAM("Could not fit!");
+    for (size_t i = 0; i < curve_points.size() - 1; i++)
+    {
+      all_sampling_points.push_back(curve_points[i]);
+      final_yaw_values.push_back(final_yaw_values.back());
+      final_actual_speeds.push_back(final_actual_speeds.back());
     }
+  }
+  
+  ROS_DEBUG("Got fit");
 
-    ROS_DEBUG("Got fit");
-
-    ROS_DEBUG_STREAM("speed_limits.size() " << speed_limits.size());
+  ROS_DEBUG_STREAM("speed_limits.size() " << speed_limits.size());
 
-    std::vector<lanelet::BasicPoint2d> sampling_points;
-    sampling_points.reserve(1 + discreet_curve.points.size() * 2);
+  std::vector<lanelet::BasicPoint2d> sampling_points;
+  sampling_points.reserve(1 + curve_points.size() * 2);
 
-    std::vector<double> distributed_speed_limits;
-    distributed_speed_limits.reserve(1 + discreet_curve.points.size() * 2);
+  std::vector<double> distributed_speed_limits;
+  distributed_speed_limits.reserve(1 + curve_points.size() * 2);
 
-    double max_x = curve_points.back().x();
-    double current_dist = 0;
-    double step_size = config_.curve_resample_step_size;
-    int current_speed_index = 0;
+  // compute subcurves to get correct time steps
+  std::vector<DiscreteCurve> sub_curves = compute_sub_curves(time_bound_points);
+  int parameter_length = 0;
+  double step_size = config_.curve_resample_step_size;
+  for (auto const discrete_curve : sub_curves)
+  {
+    double max_x = discrete_curve.points.back().point.x();
+    parameter_length += max_x / step_size;
+  }
 
-    while (current_dist < max_x - step_size) // Resample curve at tighter resolution
-    {
-      double x = current_dist;
-      double y = (*fit_curve)(x);
-      lanelet::BasicPoint2d p(x, y);
-      sampling_points.push_back(p);
+  int current_speed_index = 0;
+  int total_point_size = curve_points.size();
 
-      for (size_t i = current_speed_index; i < curve_points.size(); i++)
-      {
-        if (curve_points[i].x() >= current_dist)
-        {
-          current_speed_index = i;
-          break;
-        }
-      }
+  double next_speed_parameter_threshold = parameter_length / total_point_size;
+  double scaled_parameter = 0.0;
```

Comment:
I think I understand what this variable is, but "parameter" is a bit vague. Not really sure what a better name would be though, but maybe a comment can help make it more obvious if we can't think of a more direct name?